### PR TITLE
Integration of quantizedaffinities Branch

### PR DESCRIPTION
**Pull request by william-silversmith**
Adds support/refactoring for the following tasks:

IngestTask, HyperSquareTask, HyperSquareConsensusTask, 
MeshTask, MeshManifestTask, DownsampleTask, QuantizeAffinitiesTask, 
TransferTask, BossTransferTask, WatershedRemapTask

Adds downsampling to most tasks.

### CODE REVIEW

File: python/test/test_tasks.py [hunk 1]
```diff
@@ -126,8 +364,8 @@ def test_real_data():
     return # this is to expensive to be test by travis
     from tqdm import tqdm
     from itertools import product
-    storage = Storage('s3://neuroglancer/pinky40_v11/affinitymap-jnet')
-    scale = Precomputed(storage).info['scales'][0]
+    vol = CloudVolume('s3://neuroglancer/pinky40_v11/affinitymap-jnet')
```
Comment (tartavull) [inline python/test/test_tasks.py]: Test depending on external data

**Comment by tartavull** [comment]
I would change title of https://github.com/seung-lab/neuroglancer/pull/159/commits/5388bcee82179b43997bbcdedf446b55186cb02c
to added library `intern` to requirements.txt
The current title made me think we were missing an intern :p

**Comment by tartavull** [comment]
I wouldn't call https://github.com/seung-lab/neuroglancer/pull/159/commits/773f84b3adaeeb2ae270a2a36fbd64dbc769a04e a chore.
I think that is a refractor.
That refractor could be largely avoided if you change the code of Precomputed for the code of CloudVolume.

The only change is that CloudVolume interface takes an string and Precomputed takes an Storage object.
But you could have an if statement in the constructor to check for the types.

Given that CloudVolume complies with Precomputed interface, we could avoid breaking people tasks that hasn't being merged into master branch.

File: python/test/test_tasks.py (hunk 1 above)
Comment (tartavull) [inline python/test/test_tasks.py]: This is not that import though

Comment (wongwill86) [inline python/test/test_tasks.py]: fyi we have a test bucket s3://seunglab-test we can use too.

**Comment by jingpengw** [comment]
@william-silversmith I see you are splitting cloud-volume out, why?

**Comment by william-silversmith** [comment]
http://gph.is/2bmHPjp

**Comment by jingpengw** [comment]
@william-silversmith the queue name was hardcoded .
https://github.com/seung-lab/neuroglancer/blob/quantizedaffinities/python/neuroglancer/pipeline/task_creation.py#L485

**Comment by tartavull** [comment]
feat(ngmesh2obj): produce one file per mesh (#161) it also kind of unrelated to the rest, I would put them I'm another PR

**Comment by tartavull** [comment]
What we are trying to do is add useful features/fix bugs to what we currently have on master.
You improved our interface to download/upload volumetric data.
The interface use by all current tasks on master is called precomputed. But the interface with the changes is called cloud volume.
Currently the PR does 2 things, introduces cloudvolume and refractors all the tasks to use it. (which has the problem of introducing this breaking backward compatibility)
What I am saying is that we can have the best of both worlds.
If we merge cloudvolume but we keep the backward compatibility:
a) We get all the improvements we introduced
b) I don't need to review a refractor of all our task
c) We don't break stuff for other users.
The only drawback is that we need to make a constructor take either an string or an storage object. (which is trivial)

**Comment by tartavull** [comment]
On your point of is  Precomputed a good name or not I agree with you. Don't have my heart set on anything, agree that is a technical name that it's hard to understand unless you know about neuroglancer.
I also understand that cloudvolume is easier to understand, but be misleading for when you stored chunk in precomputed format in your local filesystem.

**Comment by william-silversmith** [comment]
ngmesh2obj was from another PR https://github.com/seung-lab/neuroglancer/pull/161

**Comment by william-silversmith** [comment]
A Precomputed shim was introduced that maintains the backwards compatibility.

**Comment by tartavull** [comment]
feel free to merge

**Comment by william-silversmith** [comment]
rebased on master to get nico's sweet requirements.txt changes

**Pull request merged**